src/storage/ResourceSet.ts

```diff
@@ -0,0 +1,15 @@
+import type { ResourceIdentifier } from '../http/representation/ResourceIdentifier';
+import type { Conditions } from './Conditions';
+
+/**
+ * A set containing resources.
+ */
+export interface ResourceSet {
+  /**
+   * Check if a resource exists in this ResourceSet.
+   * @param identifier - Identifier of resource to check.
+   *
+   * @returns A promise resolving if the resource already exists.
+   */
+  hasResource: (identifier: ResourceIdentifier, conditions?: Conditions) => Promise<boolean>;
```

Comment:
I'm not sure either actually, it's what we had but it might indeed not be necessary. We don't check the conditions in the current implementation either way so we can just drop them.

Reply:
I would drop indeed.